### PR TITLE
Fix/3

### PR DESCRIPTION
**Pull request by max246**
Implemented a filter option to hide serial logs about temperature, it uses a toggle functionality to trigger on and off.

### CODE REVIEW

**Comment by foosel** [comment]
FYI, seen the review request, it's on my TODO list, but I'm currently fully focused on 1.8.3.

File: src/components/LogView.js [hunk 1]
```diff
@@ -117,9 +141,12 @@ export default function LogView(props) {
                 matcher = line => line.includes(cs ? q : qLower);
             }
 
+
+
+
             const ind = lines.reduce((result, line, index) => {
                 if (matcher(cs ? line : line.toLowerCase())) {
-                    result.push(index);
+                        result.push(index);
```
Comment (foosel) [inline src/components/LogView.js:149]: Remove that whitespace please

File: src/components/LogView.js [hunk 2]
```diff
@@ -38,9 +38,11 @@ export default function LogView(props) {
 
     const log = props.log;
     const content = props.content;
-    const lines = content.trim().split("\n").map((line) => (line.trimEnd() + "\n"));
-    const lineCount = lines.length;
+    //const lines = content.trim().split("\n").map((line) => (line.trimEnd() + "\n"));
```
Comment (foosel) [inline src/components/LogView.js:41]: Remove this please if it's no longer needed

**Comment by foosel** [comment]
I'm a bit unsure about what this would accomplish in the context of the bundleviewer. Given that it is a debugging tool, even temperature lines are usually important, as their timing information can give infos on how the printed processes commands and how long it needs to process queries and such.

In the context of OctoPrint's terminal tab, yes, they can be unwelcome noise, but in the context of viewing `serial.log` from a system info bundle, I find them quite important for the full picture, just like every other single line. 

Can you elaborate?

**Comment by foosel** [comment]
And it seems like I'm really behind things, I totally forgot that this was a feature request. Let me raise the question on that instead.

**Comment by max246** [comment]
> And it seems like I'm really behind things, I totally forgot that this was a feature request. Let me raise the question on that instead.

Yes it was :) I just picked a while ago form the backlog list.

File: src/components/LogView.js [hunk 3]
```diff
@@ -102,6 +104,28 @@ export default function LogView(props) {
...
+            if (filter !== false && !filter(line)) {
+                filterResult.push(line);
+            } else if (filter === false) {
+                filterResult.push(line);
+            }
+        })
+
+        setLineCount(filterResult.length);
+        setLines(filterResult)
+    }
```
Comment (foosel) [inline src/components/LogView.js:125]:
```suggestion
    const onPerformFilter = (type) => {
        let filtered = false;
        if (type === "temperature") {
            filtered = line => line.match(reTemperature);
        }

        const result = content.trim().split("\n").map((line) => (line.trimEnd() + "\n")).filter(line => !filtered || !filtered(line));
        setLineCount(result.length);
        setLines(result);
    }
```

File: src/components/LogView.js [hunk 4]
```diff
@@ -117,9 +141,12 @@ export default function LogView(props) {
                 matcher = line => line.includes(cs ? q : qLower);
             }
 
+
+
+
```
Comment (foosel) [inline src/components/LogView.js:146]: Remove that whitespace please

File: src/components/LogView.js [hunk 5]
```diff
@@ -172,7 +199,15 @@ export default function LogView(props) {
...
 
-    return (
+
+        useEffect( () => {
+            let result = (content.trim().split("\n").map((line) => (line.trimEnd() + "\n")));
+            setLines(result);
+            setLineCount(result.length);
+        }, [content,props.content]);
+
+
```
Comment (foosel) [inline src/components/LogView.js:209]: Some extra empty lines here, please remove.

File: src/components/SearchBar.js [hunk 6]
```diff
@@ -105,22 +109,33 @@ export default function SearchBar(props) {
...
+              {props.showTemperatureFilter ? <Tooltip title="Toggle filter temperature"><IconButton onClick={handleToggleTemperature} color={filterTemperature ? 'primary' : 'default'} size="small"><FilterIcon /></IconButton></Tooltip> : (null)}
+              <Tooltip title="Toggle regular expression mode"><IconButton onClick={handleToggleRegex} color={regexMode ? 'primary' : 'default'} size="small"><RegexIcon /></IconButton></Tooltip>
+              <Tooltip title="Toggle case sensitivity"><IconButton onClick={handleToggleCaseSensitive} color={caseSensitive ? 'primary' : 'default'} size="small"><CaseIcon /></IconButton></Tooltip>
+              <Tooltip title="Scroll to previous match"><IconButton onClick={props.onPrev} disabled={count === 0} size="small"><PrevIcon /></IconButton></Tooltip>
+              <Tooltip title="Scroll to next match"><IconButton onClick={props.onNext} disabled={count === 0} size="small"><NextIcon /></IconButton></Tooltip>
+              <Tooltip title="Delete search query"><IconButton onClick={handleCancel} disabled={query === ""} size="small"><CancelIcon /></IconButton></Tooltip>
+            </InputAdornment>)
+    }
```
Comment (foosel) [inline src/components/SearchBar.js:138]: Why the switch to a `{ return ... }` here? 🤔

And this breaks utilization of the outer `props.onPrev`, `props.onNext` and thus the search function.

**Review by foosel — CHANGES_REQUESTED** [review]
First of all thanks for your patience, also in me clarifying the intent behind the initial request again. Now, sadly, I'm really not happy with this way of implementing it 😕 There are some direct todos further below, but also these bigger points (ordered by priority):

- The Searchbar should really not care about showing a filter button based on log names, it should just get a boolean prop set if it should show that button and have a callback for toggling it.
- I also don't like how it fully removes the lines, without any hint that they were there. OctoPrint's terminal filters inject `[...]` where stuff was filtered out (but only once per multiple filtered lines) and also indicates number of visible vs. total lines. That makes it easier for someone looking at the terminal - and here the `serial.log` - to understand that filtering is happening, and to disable that again if that was a mistake. 
- We need more flexibility in the search options here. Right now the button does one single thing, filtering out `Recv: T...` lines. That's not on par with the functionality of terminal filters present in OctoPrint. I'd prefer to have some kind of menu here to select at least the same filter options (powered by a passed property).
- Yet another issue with how filtered lines are represented: We also have line numbers here, and that makes it even more tricky. They should reflect filtered lines, so "1, 2, 3, 8, 9, 10" if lines 4-7 were filtered.  Maybe it's possible to adjust the LogLines to do all of this?

**Comment by foosel** [comment]
Decided to give this a whirl myself and implemented skipped line indicators (`[...]`), line number skipping, flexible filters, and also refactored things a bit.

**Pull request merged**